onnxruntime/core/providers/cpu/tensor/pad.cc

```diff
@@ -479,20 +519,41 @@ Status Pad::Compute(OpKernelContext* ctx) const {
 
     const Tensor& pads_tensor = *ctx->Input<Tensor>(1);
     auto pads_tensor_dims = pads_tensor.Shape().GetDims();
-    ORT_ENFORCE(pads_tensor.IsDataType<int64_t>(),
-                "Pads tensor should be an INT64 tensor");
     ORT_ENFORCE(pads_tensor_dims.size() == 1 || (pads_tensor_dims.size() == 2 && pads_tensor_dims[0] == 1),
-                "Pads tensor should be a 1D tensor of shape [2 * input_rank] "
-                "or a 2D tensor of shape [1, 2 * input_rank]");
-
+                "Pads tensor should be a 1D tensor of shape [2 * num_axes] "
+                "or a 2D tensor of shape [1, 2 * num_axes]");
     const int64_t* pads_tensor_raw_data = pads_tensor.Data<int64_t>();
     size_t pads_size = static_cast<size_t>(pads_tensor.Shape().Size());
-    ORT_ENFORCE(pads_size == 2 * data_rank,
-                "Pads tensor size should be equal to twice the input dimension count ");
-
     pads.reserve(2 * data_rank);
-    for (size_t i = 0; i < pads_size; ++i) {
-      pads.push_back(pads_tensor_raw_data[i]);
+
+    const Tensor* axes_tensor = ctx->Input<Tensor>(3);
+    if (axes_tensor) {
+      const auto& axes_tensor_dims = axes_tensor->Shape().GetDims();
+      ORT_ENFORCE(axes_tensor_dims.size() == 1, "Axes tensor should be a 1D tensor ");
```

Comment:
I know some of the code is old here and it has established a practice that it's ok to use ORT_ENFORCE in a Compute method. Since Compute already returns a Status we should try to return a Status with INVALID_ARGUMENT code.